features/update_membership_application.feature

```diff
@@ -8,31 +8,282 @@ Feature: As an Admin
+    Given I am logged in as "lars_rejected@snarkybark.se"
+    And I am on the "edit my application" page
+    And I fill in t("membership_applications.show.last_name") with "BadBadLars"
+    And I click on t("membership_applications.edit.submit_button_label")
+    Then I should see t("membership_applications.update.success")
+    And I should see status line with status t("membership_applications.pending")
+    And I am Logged out
+    And I am logged in as "admin@shf.se"
+    And I am on "LarsRejected" application page
+    Then I should see status line with status t("membership_applications.pending")
+    And I am on the "landing" page
+    And I should see 3 t("membership_applications.pending")
+    And I should see 1 t("membership_applications.waiting_for_applicant")
+    And I should see 1 t("membership_applications.accepted")
+    And I should see 0 t("membership_applications.rejected")
+
+
+  Scenario: Admin updates info thus it does not change from rejected to pending
+    Given I am on "LarsRejected" application page
+    When I click on t("membership_applications.edit_membership_application")
+    And I fill in t("membership_applications.show.last_name") with "BadBadLars"
+    And I click on t("membership_applications.edit.submit_button_label")
+    Then I should see t("membership_applications.update.success")
+    And I should see status line with status t("membership_applications.rejected")
+    When I am on the "landing" page
+    And I should see 2 t("membership_applications.pending")
+    And I should see 1 t("membership_applications.waiting_for_applicant")
+    And I should see 1 t("membership_applications.accepted")
+    And I should see 1 t("membership_applications.rejected")
+
+
+  Scenario: Admin cannot change from rejected to needs info
+    Given I am on "LarsRejected" application page
+    Then I should not see button t("membership_applications.waiting_for_applicant")
+
+
+  Scenario: Admin cannot change from rejected to accepted
+    Given I am on "LarsRejected" application page
+    When I click on t("membership_applications.accept")
+    Then I should see t("membership_applications.accept.success")
+    And I should see t("membership_applications.update.enter_member_number")
+    When I am on the "landing" page
+    Then I should see 1 t("membership_applications.waiting_for_applicant")
+    And I should see 2 t("membership_applications.pending")
+    And I should see 2 t("membership_applications.accepted")
+    And I should see 0 t("membership_applications.rejected")
+
+
+  Scenario: Admin cannot change from rejected to rejected
+    Given I am on "LarsRejected" application page
+    Then I should not see button t("membership_applications.reject")
+
 
+  Scenario: Admin cannot change from rejected to cancel needs info
+    Given I am on "LarsRejected" application page
+    Then I should not see button t("membership_applications.cancel_waiting_for_applicant")
```

Comment:
I tried to map state changes as shown in the STD to this feature.  In this snapshot, yellow highlights the state changes I did not see in the feature.  (those may be present in other feature tests - I did not look - however, it would be good to have all application state changes tested in this feature).

<img width="597" alt="screen shot 2017-01-01 at 12 07 14 pm" src="https://cloud.githubusercontent.com/assets/9968213/21582156/e19de612-d01a-11e6-964f-2896adc0ecdd.png">